3-UsingAPIs/Week2/homework/ex2-pokemonApp/index.js

```diff
@@ -18,18 +18,67 @@ Complete the four functions provided in the starter `index.js` file:
 Try and avoid using global variables. Instead, use function parameters and 
 return values to pass data back and forth.
 ------------------------------------------------------------------------------*/
-function fetchData(/* TODO parameter(s) go here */) {
+async function fetchData(url) {
   // TODO complete this function
+  const response = await fetch(url);
+  const data = await response.json();
+  return data;
 }
 
-function fetchAndPopulatePokemons(/* TODO parameter(s) go here */) {
+function fetchAndPopulatePokemons(select, data) {
   // TODO complete this function
+  const pokemons = data.results;
+  pokemons.forEach((pokemon) => {
+    const option = document.createElement('option');
+    select.appendChild(option);
+    option.value = pokemon.name;
+    option.textContent = pokemon.name;
+  });
 }
 
-function fetchImage(/* TODO parameter(s) go here */) {
+async function fetchImage(select) {
   // TODO complete this function
+  const response = await fetch(
+    `https://pokeapi.co/api/v2/pokemon/${select.value}`
+  );
+  const data = await response.json();
+  const img = document.createElement('img');
+  img.src = data.sprites.front_default;
+  document.body.appendChild(img);
+  img.style.width = '200px';
+
+  const reset = () => {
+    img.style.display = 'none';
+  };
```

Comment:
This is an interesting fix, but has one disadvantage, we will still create many `img` objects on the page which is not optimal. Yes most of them are hidden but still part of the DOM.

What we can do instead is to have only one `img` element that we create in `main` and just change it's `src` attribute every time we change the selection